docs/auth/vmware-cloud/provider.md

```diff
@@ -0,0 +1,250 @@
+
+This provider was added after the migration of the auth-backend plugin to the
+new backend system, so no default provider factory was added. Because of this,
+the installation procedure for old-style backends is slightly more involved:
+
+```ts title="packages/backend/src/plugins/auth.ts"
+import {
+  DEFAULT_NAMESPACE,
+  stringifyEntityRef,
+} from '@backstage/catalog-model';
+import {
+  createRouter,
+  providers,
+  defaultAuthProviderFactories,
+} from '@backstage/plugin-auth-backend';
+import { Router } from 'express';
+import { PluginEnvironment } from '../types';
+/* highlight-add-start */
+import {
+  commonSignInResolvers,
+  createOAuthProviderFactory,
+} from '@backstage/plugin-auth-node';
+import {
+  vmwareCloudAuthenticator,
+} from '@backstage/plugin-auth-backend-module-vmware-cloud-provider';
+/* highlight-add-end */
+
+export default async function createPlugin(
+  env: PluginEnvironment,
+): Promise<Router> {
+  return await createRouter({
+    logger: env.logger,
+    config: env.config,
+    database: env.database,
+    discovery: env.discovery,
+    tokenManager: env.tokenManager,
+    providerFactories: {
+      ...defaultAuthProviderFactories,
+      /* highlight-add-start */
+      vmwareCloudServices: createOAuthProviderFactory({
+        authenticator: vmwareCloudAuthenticator,
+        signInResolver:
+          commonSignInResolvers.emailLocalPartMatchingUserEntityName(),
+      }),
+      /* highlight-add-end */
+```
+
+In the above, `commonSignInResolvers.emailLocalPartMatchingUserEntityName()`
+can be replaced with a more suitable resolver for the app in question.
+
+## Configure Sign-in Resolution
+
+See [Sign-in Identities and Resolvers](../identity-resolver.md) for details.
+
+## Create a Utility API
+
+At the moment, this auth provider is not included in the default API factories,
+but you can create one with a change to your frontend like:
+
+```ts title="packages/app/src/apis.ts"
+import {
+  ScmIntegrationsApi,
+  scmIntegrationsApiRef,
+  ScmAuth,
+} from '@backstage/integration-react';
+import {
+  costInsightsApiRef,
+  ExampleCostInsightsClient,
+} from '@backstage/plugin-cost-insights';
+import {
+  graphQlBrowseApiRef,
+  GraphQLEndpoints,
+} from '@backstage/plugin-graphiql';
+/* highlight-add-start */
+import { OAuth2 } from '@backstage/core-app-api';
+/* highlight-add-end */
+import {
+  AnyApiFactory,
+  /* highlight-add-start */
+  ApiRef,
+  OpenIdConnectApi,
+  ProfileInfoApi,
+  BackstageIdentityApi,
+  SessionApi,
+  /* highlight-add-end */
+  configApiRef,
+  /* highlight-add-start */
+  createApiRef,
+  /* highlight-add-end */
+  createApiFactory,
+  discoveryApiRef,
+  errorApiRef,
+  fetchApiRef,
+  githubAuthApiRef,
+  /* highlight-add-start */
+  oauthRequestApiRef,
+  /* highlight-add-end */
+} from '@backstage/core-plugin-api';
+import { AuthProxyDiscoveryApi } from './AuthProxyDiscoveryApi';
+
+/* highlight-add-start */
+export const vmwareCloudAuthApiRef: ApiRef<
```

Comment:
Is this provider only useful for sign-in or is  there some OAuth access delegation going on as well that would be useful to provide to plugins? Asking mainly for future work

Either way let's go ahead and add both this API ref and the factory the to the core APIs. `core-plugin-api` + `frontend-plugin-api` + `app-defaults`

Long term I'm looking to get rid of the need for API Refs for sign-in, and possibly consolidating all OAuth providers into a single API. But can't promise that's gonna be done soon enough to close this gap

Reply:
There is a VMware-specific use case that will require access tokens for use with an external service, so "yes" to access delegation. I'll file an issue/PR for adding this utility API to the core.